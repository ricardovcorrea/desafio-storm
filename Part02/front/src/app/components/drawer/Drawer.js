import React from "react";
import "./Drawer.scss";

import Drawer from "@material-ui/core/Drawer";

const CustomDrawer = props => {
  return (
    <Drawer anchor="right" open={props.open} onClose={props.onClose}>
      Drawer content
    </Drawer>
  );
};

export default CustomDrawer;
