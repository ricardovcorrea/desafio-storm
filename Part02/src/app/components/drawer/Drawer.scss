.drawer {
  width: 500px !important;
  background-color: #f5f5f5;
  height: 100%;
}

.drawerHeader {
  background-color: white;
  padding: 30px;
  align-items: center;
  display: flex;
  height: 50px;
}

.drawerHeaderIcon {
  color: #a1a1a1;
}

.drawerHeaderTitle {
  margin-left: 15px;
  font-size: 20px;
}

.drawerBody {
  padding: 30px;
  padding-top: 40px;
}

.drawerText {
  font-size: 18px;
  font-style: italic;
  color: #a1a1a1;
}

.horizontalDivider {
  width: 20px;
  height: 3px;
  background-color: #e3e3e3;
  margin-top: 15px;
}

.filterBox {
  border-bottom: 2px solid #e3e3e3;
  display: flex;
  flex-direction: row;
  padding-top: 30px;
  padding-bottom: 30px;
  justify-content: space-between;
}

.filterIcon {
  margin-right: 15px;
  color: #a1a1a1;
}

.filterText {
  display: flex;
  flex-direction: row;
  color: rgb(216, 51, 103);
  align-items: center;
  font-size: 18px;
  font-weight: 500;
}

.applyButton {
  height: 70px;
  border: 1px solid rgb(216, 51, 103);
  color: rgb(216, 51, 103);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 3px;
  margin-top: 40px;
  cursor: pointer;
  opacity: 0.5;
}
