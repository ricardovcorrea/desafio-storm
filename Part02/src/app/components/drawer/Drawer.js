import React from "react";
import "./Drawer.scss";

import Drawer from "@material-ui/core/Drawer";
import Icon from "@material-ui/core/Icon";
import Button from "../button/Button";

const CustomDrawer = props => {
  return (
    <Drawer anchor="right" open={props.open} onClose={props.onClose}>
      <div className="drawer">
        <div className="drawerHeader">
          <Icon className="drawerHeaderIcon">tune</Icon>
          <span className="drawerHeaderTitle">FILTROS</span>
          <div className="grow"></div>
          <Button onClick={() => props.onClose()} icon="close"></Button>
        </div>
        <div className="drawerBody">
          <span className="drawerText">
            Utilize os filtros abaixo para refinar os resultados da tabela,
            clique no botão APLICAR para salvar as alterações.
          </span>
          <div className="horizontalDivider">&nbsp;</div>
          <div className="filterBox">
            <div className="filterText">
              <Icon className="filterIcon">date_range</Icon>
              <div>TODAS AS DATAS DE INCLUSÃO</div>
            </div>
            <Icon className="drawerHeaderIcon">arrow_drop_down</Icon>
          </div>
          <div className="filterBox">
            <div className="filterText">
              <Icon className="filterIcon">date_range</Icon>
              <div>TODAS AS DATAS DE ALTERAÇÃO</div>
            </div>
            <Icon className="drawerHeaderIcon">arrow_drop_down</Icon>
          </div>
          <div className="filterBox">
            <div className="filterText">
              <Icon className="filterIcon">date_range</Icon>
              <div>ATIVOS E INATIVOS</div>
            </div>
            <Icon className="drawerHeaderIcon">arrow_drop_down</Icon>
          </div>
          <div onClick={() => props.onClose()} className="applyButton">
            APLICAR
          </div>
        </div>
      </div>
    </Drawer>
  );
};

export default CustomDrawer;
