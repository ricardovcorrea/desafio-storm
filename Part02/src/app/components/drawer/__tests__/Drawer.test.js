import React from 'react';
import { render } from '@testing-library/react';
import Drawer from '../Drawer';

test('renders drawer', () => {
  const container = render(<Drawer />)
  expect(container.container).toMatchSnapshot()
});
