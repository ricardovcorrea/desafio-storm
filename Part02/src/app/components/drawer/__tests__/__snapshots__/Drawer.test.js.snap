// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders drawer 1`] = `
<div
  aria-hidden="true"
/>
`;
